src/components/Visualization/useAnalyticsData.js

```diff
@@ -136,6 +139,7 @@ const useAnalyticsData = ({
     page,
     pageSize,
 }) => {
+    const uiState = useSelector(sGetUi)
```

Comment:
can we avoid passing the whole `ui` object around and only pick out the parts that are actually used? (looks like `program` and `input`)

Reply:
also, you already have `visualization` (which is `current`, right?), wouldn't that include the parts needed anyway?

Reply:
![image](https://user-images.githubusercontent.com/12590483/151166752-d5f2d8b2-284f-4f6b-9013-5fa05ff3c490.png)
debugging this I see that `program` and `input` are missing from `visualization`, but that's just because they haven't been added yet, right? so once they're added to `visualization` you could just access everything that `uiState` provides through that prop?

Reply:
This was an excellent point. The fact that these values were not present in the first place was the problem, so reading them directly from the UI state was not really a solution. In 7d50f1b I have ensured that the required properties get added to `current` from `ui`.